test/RequestDetailsTests.cpp

```diff
@@ -793,21 +793,21 @@ void RequestDetailsTests::testAuthorization()
     auth3.authorizeRequest(req3);
     LOK_ASSERT_EQUAL(std::string("Basic huhu=="), req3.get("Authorization"));
 
-    Authorization auth4(Authorization::Type::Header, "  Authorization: Basic blah== \n\rX-Something:   additional  ");
+    Authorization auth4(Authorization::Type::Header, "  Authorization: Basic blah== \n\rX-Something:   additional  ", false);
     Poco::Net::HTTPRequest req4;
     auth4.authorizeRequest(req4);
     LOK_ASSERT_MESSAGE("Exected request to have Authorization header", req4.has("Authorization"));
     LOK_ASSERT_EQUAL(std::string("Basic blah=="), req4.get("Authorization"));
     LOK_ASSERT_MESSAGE("Exected request to have X-Something header", req4.has("X-Something"));
     LOK_ASSERT_EQUAL(std::string("additional"), req4.get("X-Something"));
 
-    Authorization auth5(Authorization::Type::Header, "  Authorization: Basic huh== \n\rX-Something-More:   else  \n\r");
+    Authorization auth5(Authorization::Type::Header, "  Authorization: Basic huh== \n\rX-Something-More:   else  \n\r", false);
     Poco::Net::HTTPRequest req5;
     auth5.authorizeRequest(req5);
     LOK_ASSERT_EQUAL(std::string("Basic huh=="), req5.get("Authorization"));
     LOK_ASSERT_EQUAL(std::string("else"), req5.get("X-Something-More"));
 
-    Authorization auth6(Authorization::Type::None, "Authorization: basic huh==");
+    Authorization auth6(Authorization::Type::None, "Authorization: basic huh==", false);
```

Comment:
Did you consider adding a test case in this suite where the Authorization instance is created with noHeader = true?

I would expect you could assert that the request has no "Authorization" header in that case.